### ISSUE TITLE
Namespace stuck deleting

### ISSUE THREAD

**Issue by Dirbaio** [kubedb/project#759]
Kubernetes on GKE version v1.15.11-gke.13
KubeDB version kubedb/operator:v0.13.0-rc.0

1. `kubectl create namespace myns`
2. `kubectl -n myns apply -f postgres.yaml`

```yaml
apiVersion: kubedb.com/v1alpha1
kind: Postgres
metadata:
  name: backend-postgres
spec:
  version: "11.1"
  storageType: Durable
  storage:
    storageClassName: "faster"
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 1Gi
  terminationPolicy: WipeOut
```

3. Wait for the database to come up
4. `kubectl delete namespace myns`
5. Namespace stays stuck in `Terminating` state forever. KubeDB operator prints the following:

```
E0525 14:45:01.838408       1 workqueue.go:36] error in wiping out database.: error in getting db secret: secrets "backend-postgres-auth" not found
E0525 14:45:01.838433       1 worker.go:76] Failed to process key myns/backend-postgres. Reason: error in wiping out database.: error in getting db secret: secrets "backend-postgres-auth" not found
E0525 14:45:01.838450       1 worker.go:90] error in wiping out database.: error in getting db secret: secrets "backend-postgres-auth" not found
I0525 14:45:01.838489       1 worker.go:91] Dropping key "myns/backend-postgres" out of the queue: error in wiping out database.: error in getting db secret: secrets "backend-postgres-auth" not found
```

I think what's happening is Kubernetes is deleting the secret before kubedb gets to delete the database. Why does WipeOut require the secret anyway? Shouldn't erasing the pods and volumes be enough?

**Comment by sjagoe** [comment]
I also have this error, and I can't create a secret to work around the issue because the namespace is in the `Terminating` state.

Is there any way to have the operator force-delete a resource?

**Comment by sjagoe** [comment]
Found a workaround here: https://github.com/kubedb/project/issues/684#issuecomment-553314241

**Comment by Dirbaio** [comment]
I confirm the workaround works! Thanks

**Comment by tamalsaha** [comment]
This issue has been resolved in KubeDB release v2021.03.17 . Please open a new bug if you still see this issue.
